tobac/merge_split.py

```diff
@@ -0,0 +1,477 @@
+    We allocate a minimum number of dimensions as variables with dimensions
+    under the minimum value do not benefit from tangibly from this encoding.
+
+    Parameters
+    ----------
+    nc_grids : xarray.core.dataset.Dataset
+        Xarray dataset that is intended to be exported as netcdf
+
+    min_dims : integer
+        The minimum number of dimesnions, in integer value, a variable must have in order
+        set the netcdf compression encoding.
+
+    Returns
+    -------
+    nc_grids : xarray.core.dataset.Dataset
+        Xarray dataset with netcdf compression encoding for variables with two (2) or more dimensions
+
+    """
+
+    for var in nc_grids:
+        if len(nc_grids[var].dims) >= min_dims:
+            # print("Compressing ", var)
+            nc_grids[var].encoding["zlib"] = True
+            nc_grids[var].encoding["complevel"] = 4
+            nc_grids[var].encoding["contiguous"] = False
+    return nc_grids
+
+
+def standardize_track_dataset(TrackedFeatures, Mask, Projection=None):
+    """
+    Combine a feature mask with the feature data table into a common dataset.
+
+    returned by tobac.themes.tobac_v1.segmentation
+    with the TrackedFeatures dataset returned by tobac.themes.tobac_v1.linking_trackpy.
+
+    Also rename the variables to be more desciptive and comply with cf-tree.
+
+    Convert the default cell parent ID  to an integer table.
+
+    Add a cell dimension to reflect
+
+    Projection is an xarray DataArray
+
+    TODO: Add metadata attributes
+
+    Parameters
+    ----------
+    TrackedFeatures : xarray.core.dataset.Dataset
+        xarray dataset of tobac Track information, the xarray dataset returned by tobac.tracking.linking_trackpy
+
+    Mask: xarray.core.dataset.Dataset
+        xarray dataset of tobac segmentation mask information, the xarray dataset returned
+        by tobac.segmentation.segmentation
+
+
+    Projection : xarray.core.dataarray.DataArray, default = None
+        array.DataArray of the original input dataset (gridded nexrad data for example).
+        If using gridded nexrad data, this can be input as: data['ProjectionCoordinateSystem']
+        An example of the type of information in the dataarray includes the following attributes:
+        latitude_of_projection_origin :29.471900939941406
+        longitude_of_projection_origin :-95.0787353515625
+        _CoordinateTransformType :Projection
+        _CoordinateAxes :x y z time
+        _CoordinateAxesTypes :GeoX GeoY Height Time
+        grid_mapping_name :azimuthal_equidistant
+        semi_major_axis :6370997.0
+        inverse_flattening :298.25
+        longitude_of_prime_meridian :0.0
+        false_easting :0.0
+        false_northing :0.0
+
+    Returns
+    -------
+
+    ds : xarray.core.dataset.Dataset
+        xarray dataset of merged Track and Segmentation Mask datasets with renamed variables.
+
+    """
+    feature_standard_names = {
```

Comment:
I think it's possible that these would be better off in a JSON. That would allow us to pull from there to build the documentation, too. However, I don't think that's necessary for this PR.